crates/scroll/evm/src/execute.rs

```diff
@@ -0,0 +1,545 @@
+}
+
+/// The factory for a [`ScrollExecutionStrategy`].
+#[derive(Clone, Debug)]
+pub struct ScrollExecutionStrategyFactory<EvmConfig = ScrollEvmConfig> {
+    /// The chain specification for the [`ScrollExecutionStrategy`].
+    chain_spec: Arc<ScrollChainSpec>,
+    /// The Evm configuration for the [`ScrollExecutionStrategy`].
+    evm_config: EvmConfig,
+}
+
+impl ScrollExecutionStrategyFactory {
+    /// Returns a new instance of the [`ScrollExecutionStrategyFactory`].
+    pub fn new(chain_spec: Arc<ScrollChainSpec>, scroll_config: ScrollChainConfig) -> Self {
+        let evm_config = ScrollEvmConfig::new(chain_spec.clone(), scroll_config);
+        Self { chain_spec, evm_config }
+    }
+}
+
+impl<EvmConfig> BlockExecutionStrategyFactory for ScrollExecutionStrategyFactory<EvmConfig>
+where
+    EvmConfig: ConfigureEvm<Header = Header>,
+{
+    /// Associated strategy type.
+    type Strategy<DB: Database<Error: Into<ProviderError> + Display>>
+        = ScrollExecutionStrategy<DB, EvmConfig>
+    where
+        State<DB>: FinalizeExecution<Output = BundleState>;
+
+    /// Creates a strategy using the give database.
+    fn create_strategy<DB>(&self, db: DB) -> Self::Strategy<DB>
+    where
+        DB: Database<Error: Into<ProviderError> + Display>,
+        State<DB>: FinalizeExecution<Output = BundleState>,
+    {
+        let state =
+            State::builder().with_database(db).without_state_clear().with_bundle_update().build();
+        ScrollExecutionStrategy::new(self.chain_spec.clone(), self.evm_config.clone(), state)
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::{ScrollEvmConfig, ScrollExecutionStrategy, ScrollExecutionStrategyFactory};
+    use reth_chainspec::MIN_TRANSACTION_GAS;
+    use reth_evm::execute::ExecuteOutput;
+    use reth_primitives::{Block, BlockBody, BlockWithSenders, Receipt, TransactionSigned, TxType};
+    use reth_primitives_traits::transaction::signed::SignedTransaction;
+    use reth_scroll_chainspec::ScrollChainSpecBuilder;
+    use reth_scroll_consensus::{
+        BLOB_SCALAR_SLOT, COMMIT_SCALAR_SLOT, CURIE_L1_GAS_PRICE_ORACLE_BYTECODE,
+        CURIE_L1_GAS_PRICE_ORACLE_STORAGE, IS_CURIE_SLOT, L1_BASE_FEE_SLOT, L1_BLOB_BASE_FEE_SLOT,
+        L1_GAS_PRICE_ORACLE_ADDRESS, OVER_HEAD_SLOT, SCALAR_SLOT,
+    };
+    use revm::{
+        db::states::{bundle_state::BundleRetention, StorageSlot},
+        primitives::{Address, B256, U256},
+        shared::AccountInfo,
+        Bytecode, EmptyDBTyped, TxKind,
+    };
+
+    fn strategy() -> ScrollExecutionStrategy<EmptyDBTyped<ProviderError>, ScrollEvmConfig> {
+        let chain_spec = Arc::new(ScrollChainSpecBuilder::scroll_mainnet().build());
+        let config = ScrollChainConfig::mainnet();
+        let factory = ScrollExecutionStrategyFactory::new(chain_spec, config);
+        let db = EmptyDBTyped::<ProviderError>::new();
+
+        factory.create_strategy(db)
+    }
+
+    const BLOCK_GAS_LIMIT: u64 = 10_000_000;
```

Comment:
Do we have this constant set anywhere (not in test code) that we can leverage?

Reply:
unfortunately not, I thought maybe in the `reth-scroll-chainspec` but it doesn't look like it.